frontend/src/Components/Navbar/Navbar.tsx

```diff
@@ -1,47 +1,70 @@
-import classNames from 'classnames';
+import { Icon } from '@iconify/react';
+import { default as classNames } from 'classnames';
+import { useCallback, useEffect, useState } from 'react';
 import { useTranslation } from 'react-i18next';
-import { NavLink as Link, useNavigate } from 'react-router-dom';
+import { NavLink as Link, useLocation, useNavigate } from 'react-router-dom';
 import { logout } from '~/api';
-import { englishFlag, logoWhite, norwegianFlag, profileIcon } from '~/assets';
+import { englishFlag, logoBlack, logoWhite, norwegianFlag } from '~/assets';
 import { useAuthContext } from '~/AuthContext';
 import { Button, ThemeSwitch } from '~/Components';
+import { THEME } from '~/constants';
 import { useDesktop } from '~/hooks';
 import { STATUS } from '~/http_status_codes';
 import { KEY, LANGUAGES } from '~/i18n/constants';
 import { ROUTES } from '~/routes';
-import styles from './Navbar.module.scss';
 import { useGlobalContext } from '../../GlobalContextProvider';
+import styles from './Navbar.module.scss';
+
+const scrollDistanceForOpaque = 30;
 
 export function Navbar() {
+  const { theme } = useGlobalContext();
+  const isDarkTheme = theme === THEME.DARK;
   const { mobileNavigation, setMobileNavigation } = useGlobalContext();
   const { t, i18n } = useTranslation();
   const { user, setUser } = useAuthContext();
   const navigate = useNavigate();
   const isDesktop = useDesktop();
 
+  // Scroll detection
+  const [scrollY, setScrollY] = useState(window.scrollY);
+  const scrolledNavbar = scrollY > scrollDistanceForOpaque;
+  const handleNavigation = useCallback((e: Event) => {
+    const window = e.currentTarget;
+    if (window != null) {
+      setScrollY(window.scrollY);
+    }
+  }, []);
+  useEffect(() => {
+    window.addEventListener('scroll', handleNavigation);
+    return () => {
+      window.removeEventListener('scroll', handleNavigation);
+    };
+  }, [handleNavigation]);
+
+  // Navbar style
+  const transparentNavbar = useLocation().pathname == '/' && !scrolledNavbar && !mobileNavigation;
+  const navbarStyle = classNames(
+    transparentNavbar && styles.transparent_navbar,
+    mobileNavigation && styles.navbar_mobile,
+  );
+  const navbarImage = isDarkTheme ? logoWhite : transparentNavbar ? logoWhite : logoBlack;
```

Comment:
Usikker på om denne er riktig, men kan vi slippe dobbel ternary?
```suggestion
  const navbarImage = isLightTheme && transparentNavbar? logoWhite : logoBlack;
```